internal/security/secretstore/init.go

```diff
@@ -159,6 +172,32 @@ func (b *Bootstrap) Handler(
 	gk := NewGokeyGenerator(absPath)
 	loggingClient.Warn("WARNING: The gokey generator is a reference implementation for credential generation and the underlying libraries not been reviewed for cryptographic security. The user is encouraged to perform their own security investigation before deployment.")
 	cred := NewCred(req, absPath, gk, configuration.SecretService.GetSecretSvcBaseURL(), loggingClient)
+
+	// Detour: Create secrets engine if not there already
+	vaultToken, err := GetAccessToken(absPath)
+	if err != nil {
+		loggingClient.Error(fmt.Sprintf("failed to reload root token: %s", err.Error()))
+		os.Exit(1)
+	}
+	// Note: trailing slash on secret/ is required.
+	installed, err := vc.CheckSecretEngineInstalled(vaultToken, "secret/", "kv")
+	if err != nil {
+		loggingClient.Error(fmt.Sprintf("failed call to check if kv secrets engine is insalled: %s", err.Error()))
+		os.Exit(1)
+	}
+	if !installed {
+		loggingClient.Info("enabling kv secrets engine for the first time...")
```

Comment:
Could we stick to "kv secrets engine" or "kv secrets store". Both being used within the span of a few lines to refer to the same thing

Reply:
Changed to "KV secrets engine" everywhere.